File: invision-mobile/lib/src/core/enums/store_rank.dart
enum StoreRank {
  platinum,
  gold,
  silver,
  bronze,
  unranked;

  String get label => switch (this) {
        platinum => 'Platinum',
        gold => 'Gold',
        silver => 'Silver',
        bronze => 'Bronze',
        unranked => 'Unranked',
      };

  static StoreRank fromString(String value) => StoreRank.values.firstWhere(
        (e) => e.name == value,
        orElse: () => StoreRank.unranked,
      );
}

File: invision-backend/app/Enums/StoreRank.php
<?php

namespace App\Enums;

enum StoreRank: string
{
    case Platinum = 'platinum';
    case Gold = 'gold';
    case Silver = 'silver';
    case Bronze = 'bronze';
    case Unranked = 'unranked';

    public function label(): string
    {
        return match ($this) {
            self::Platinum => 'Platinum',
            self::Gold => 'Gold',
            self::Silver => 'Silver',
            self::Bronze => 'Bronze',
            self::Unranked => 'Unranked',
        };
    }
}
